syntax = "proto3";
package webSiteGRPC;
option go_package = "videoDynamicAcquisition/proto;proto";

message userInfo{
  map<string, string> cookies = 1;
  string lastHistoryTime = 2;
  string lastUpdateTime = 3;
}
message videoInfoResponse{
  string title = 1;  // 视频标题
  string desc = 2; // 视频描述
  string cover = 3; // 视频封面
  string uid = 4; // 视频uid,在该网站的唯一标识
  int64 duration = 5; // 视频时长
  int64 updateTime = 6; // 视频上传时间
  int64 collectTime = 23; // 指定用户收藏该视频时间
  repeated tagInfoResponse tags = 7; // 视频标签
  classifyInfoResponse classify = 8; // 视频分区分类
  int64 viewNumber = 9; // 观看人数
  int64 danmaku = 10; // 视频弹幕数
  int64 reply = 11; // 视频评论数
  int64 favorite = 12; // 视频收藏数
  int64 coin = 13; // 视频硬币
  int64 share = 14; // 视频分享
  int64 nowRank = 15; // 当前排名
  int64 hisRank = 16; // 历史排名
  int64 like = 17; // 点赞
  int64 dislike = 18; // 点踩
  string evaluation = 19; // 评分
  repeated AuthorInfoResponse authors = 20; // 视频作者
  viewInfoResponse viewInfo = 21; // 观看信息
  string webSiteName = 22; // 网站名称
  bool IsInvalid = 24; // 是否失效
  int32 errorCode = 50; // 错误码
  string errorMsg = 51; // 错误信息
  string requestUserName = 52; // 请求的用户名
  int64 requestUserId = 53; // 请求的用户id
  int64 webSiteId = 54; // 网站id
}
message AuthorInfoResponse{
  string author = 1; // 多人投稿的时候参与身份信息
  string name = 2;
  string avatar = 3;
  string desc = 4;
  string uid = 5;
  uint64 followNumber = 6;
  int64 followTime = 7;
  string webSiteName = 22;
  int32 errorCode = 50;
  string errorMsg = 51;
  string requestUserName = 52;
  int64 requestUserId = 53;
  int64 webSiteId = 54;
}
message tagInfoResponse{
  string name = 1;
  int64 id = 2;
  int64 tagType = 3; // 1:普通标签 2:bgm 3：推广标签 4:活动
  int32 errorCode = 50;
  string errorMsg = 51;
}
message classifyInfoResponse{
  string name = 1;
  int64 id = 2;
}
message viewInfoResponse{
  int64 viewTime = 1;
  int64 viewDuration = 2;
}
message collectionInfoResponse{
  string name = 1;
  string uid = 2;
  AuthorInfoResponse author = 3; // 创建者信息
  int64 creatTime = 4;
  int64 appendTime = 5; // 视频添加到合集的时间
  int32 errorCode = 50;
  string errorMsg = 51;
}

message collectionInfo{
  string collectionType = 1; // folder 个人收藏夹  2 订阅他人的收藏夹
  int64 collectionId = 2; // 网站的收藏夹id
  int64 collectionVideoCount = 3; // 收藏夹的视频数量
  int64 lastAppendTime = 4; // 最后一次添加视频的时间
  string name = 5; // 收藏夹名称
  string upperName = 6; // 收藏夹创建者
  string upperUid = 7; // 收藏夹创建者uid
  repeated videoInfoResponse video = 20;
  int64 videoNumber = 21;

  // 请求信息
  string webSiteName = 49;
  int32 errorCode = 50;
  string errorMsg = 51;
  string requestUserName = 52;
  int64 requestUserId = 53;
  int64 webSiteId = 54;
}


message collectionInfoRequest{
  userInfo user = 1;
  repeated collectionInfo collection = 2;
}

message getVideoListRequest{
  userInfo userInfo = 1;
  string videoIdList = 2;
}

message VideoDetailResponse{
  videoInfoResponse videoDetail = 1;
  repeated videoInfoResponse recommendVideo = 2;
}

service WebSiteService {
  rpc GetUserFollowUpdate(userInfo) returns (stream  videoInfoResponse);
  rpc GetUserViewHistory(userInfo) returns (stream  videoInfoResponse);
  rpc GetUserFollowList(userInfo) returns (stream AuthorInfoResponse);
  rpc GetUserCollectionList(collectionInfoRequest) returns (stream collectionInfo);
  rpc GetHotVideoList(getVideoListRequest) returns (stream  videoInfoResponse);
  rpc GetSelfInfo(userInfo) returns (AuthorInfoResponse);
  rpc GetWaitWatchList(userInfo)returns (collectionInfo);

  // 获取基础信息。以下接口全部使用代理发送请求
  rpc GetVideoDetail(stream getVideoListRequest) returns (stream  VideoDetailResponse);
}